ratpack-pac4j/src/main/java/ratpack/pac4j/RatpackPac4j.java

```diff
@@ -173,8 +178,75 @@ public static Handler authenticator(String path, ClientsProvider clientsProvider
+   * If there is no user profile present in the session (i.e. user not logged in), authentication will be initiated based on the given client type (i.e. redirect to the {@link #authenticator(Client[])} handler).
+   * If there is a {@link UserProfile}, authorizations are checked based on the provided {@link Authorizer}: if the validation fails, a 403 error page is returned, otherwise we delegate downstream (the user profile is pushed into the context registry).
+   * <p>
+   * This handler requires a {@link Clients} instance available in the context registry.
+   * As such, this handler should be downstream of the {@link #authenticator(Client[])} handler.
+   *
+   * <pre class="java">{@code
+   * import org.pac4j.core.profile.UserProfile;
+   * import org.pac4j.http.client.indirect.IndirectBasicAuthClient;
+   * import org.pac4j.http.credentials.authenticator.test.SimpleTestUsernamePasswordAuthenticator;
+   * import ratpack.guice.Guice;
+   * import ratpack.pac4j.RatpackPac4j;
+   * import ratpack.session.SessionModule;
+   * import ratpack.test.embed.EmbeddedApp;
+   *
+   * import static org.junit.Assert.assertEquals;
+   *
+   * public class Example {
+   *   public static void main(String... args) throws Exception {
+   *     EmbeddedApp.of(s -> s
+   *       .registry(Guice.registry(b -> b.module(SessionModule.class)))
+   *       .handlers(c -> c
+   *         .all(RatpackPac4j.authenticator(new IndirectBasicAuthClient(new SimpleTestUsernamePasswordAuthenticator())))
+   *         .prefix("require-authz", a -> a
+   *           .all(RatpackPac4j.security(IndirectBasicAuthClient.class, (ctx, profile) -> { return "user".equals(profile.getId()); }))
+   *           .get(ctx -> ctx.render("Hello " + ctx.get(UserProfile.class).getId()))
+   *         )
+   *         .get("logout", ctx -> RatpackPac4j.logout(ctx).then(() -> ctx.redirect("/")))
+   *         .get(ctx -> ctx.render("no auth required"))
+   *       )
+   *     ).test(httpClient -> {
+   *       httpClient.requestSpec(r -> r.redirects(1));
+   *       assertEquals("no auth required", httpClient.getText());
+   *       assertEquals(401, httpClient.get("require-authz").getStatusCode());
+   *       assertEquals(403, httpClient.requestSpec(r -> r.basicAuth("u", "u")).get("require-authz").getStatusCode());
+   *       httpClient.get("logout");
+   *       assertEquals("Hello user", httpClient.requestSpec(r -> r.basicAuth("user", "user")).getText("require-authz"));
+   *     });
+   *   }
+   * }
+   * }</pre>
+   *
+   * @param clientType the client type to use to authenticate with if required
+   * @param authorizers the authorizers to check authorizations
+   * @return a handler
+   */
+  @SafeVarargs
+  public static Handler security(Class<? extends Client<?, ?>> clientType, Authorizer<UserProfile>... authorizers) {
```

Comment:
About the `Authorizer<? extends UserProfile>`, I can change that but as the `login` method returns a `UserProfile`, I have an unchecked cast on the user profile: `a.isAuthorized(webContext, (T) userProfile)))`: not a big deal from my point of view. Otherwise, I need to pass the profile type.

What do you think?